### PR TITLE
fix: update Popover guidance and examples to show how to set focus on static popovers

### PR DESCRIPTION
**Pull request by smhigley**
Previously, our Popover examples without nested interactive items weren't getting focus when opened. This PR does two things:

1. Updates those examples to add `tabIndex={-1}` to the `PopoverSurface`
2. Adds Best Practices guidance to tell authors to do so

I also updated the example with changing content to prevent focus from being lost.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Avatar-mount_base.html">642</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Avatar-mount.html">631</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Button-mount_base.html">322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Button-mount.html">312</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Field-mount_base.html">1126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Field-mount.html">1143</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProvider-mount_base.html">690</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProvider-mount.html">712</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProviderWithTheme-mount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProviderWithTheme-mount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">60</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">59</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">87</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/InfoButton-mount_base.html">16</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/InfoButton-mount.html">12</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/MakeStyles-mount_base.html">853</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/MakeStyles-mount.html">866</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Persona-mount_base.html">1751</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/Persona-mount.html">1715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/SpinButton-mount_base.html">1394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29556/perf-test-react-components/SpinButton-mount.html">1381</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ export const AnchorToCustomTarget = () => {
           <Button>Popover trigger</Button>
         </PopoverTrigger>
 
-        <PopoverSurface>
+        <PopoverSurface tabIndex={-1}>
```
Comment (kolaps33) [inline packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:44]: This looks as good change to me. 

But when I tried with NVDA, it narrates the popover as "clickable" (what is not correct in this case):
image

what I saw in the past, it is some "feature" of NVDA it narrates "clickable" when there is "onclick" listener on the element...
I think we shouldn't have the "onclick" listener on not actionable element... 

this "clickable" narrated by NVDA can be pretty confusing for the user...

Comment (ling1726) [reply packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:44]: Let's create an issue for this separately to track

Comment (smhigley) [reply packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:44]: This is an NVDA thing that is widespread -- an NVDA user would almost certainly be so used to this that it wouldn't be an issue

**Pull request merged**